src/main/java/com/yahoo/bullet/bql/parser/ASTBuilder.java

```diff
@@ -551,7 +562,11 @@ private Number getSignedNumber(BQLBaseParser.NumberContext context) {
         String value = context.value.getText();
         switch (context.value.getType()) {
             case BQLBaseLexer.INTEGER_VALUE:
-                return negative ? -Integer.valueOf(value) : Integer.valueOf(value);
+                try {
+                    return negative ? -Integer.valueOf(value) : Integer.valueOf(value);
+                } catch (NumberFormatException e) {
```

Comment:
For big numbers?

Reply:
Convenient for big numbers. Will obviously still crash and give an uninterpretable exception if the number is too too large though ("For input: your unparseable number here")